### ISSUE TITLE
As a developer, I would like to only save confirmed datamodel objects

### ISSUE THREAD

**Issue by randypitcherii** [hashmapinc/Tempus#571]
Objects are currently saved before confirmation in order to obtain a GUID, but that is bad behavior and results in bugs if someone cancels after adding an object.

**Comment by SPParnerkar** [comment]
I noticed an issue when working with Data Model UI:

When you add Data Model Objects, the title in top left corner changes to 'Dummy Data Model'. It needs to be edited again. This can be avoided by displaying the Title entered in Create Data Model modal box. Also, it should not be editable.

If its fixed or covered elsewhere , please ignore this comment.

**Comment by randypitcherii** [comment]
Can you please post a screenshot?

**Comment by randypitcherii** [comment]
I'm not able to recreate the issue.

I searched for 'Dummy Data Model' in the UI code base and found nothing as well.

I had been using that term in an older version of the data model editor, but that was a couple PRs ago. Are you sure you're on the latest version of dev?

**Comment by SPParnerkar** [comment]
Yes, I was running an older version. It works fine on latest. Thanks!
